python-avd/pyavd/_eos_designs/structured_config/base/platform_mixin.py

```diff
@@ -0,0 +1,133 @@
+    def platform(self: AvdStructuredConfigBaseProtocol) -> None:
+        """
+        Set platform keys in structured config based on platform data-model, using various platform-related information.
+
+        The following settings are used:
+
+        * platform_settings.lag_hardware_only,
+        * platform_settings.trident_forwarding_table_partition and switch.evpn_multicast facts
+        * data_plane_cpu_allocation_max,
+        * platform_settings.feature_support.platform_sfe_interface_profile.supported
+        """
+        if (lag_hardware_only := self.shared_utils.platform_settings.lag_hardware_only) is not None:
+            self.structured_config.platform.sand.lag.hardware_only = lag_hardware_only
+
+        trident_forwarding_table_partition = self.shared_utils.platform_settings.trident_forwarding_table_partition
+        if trident_forwarding_table_partition and self.shared_utils.evpn_multicast:
+            self.structured_config.platform.trident.forwarding_table_partition = trident_forwarding_table_partition
+
+        if (cpu_max_allocation := self.shared_utils.node_config.data_plane_cpu_allocation_max) is not None:
+            self.structured_config.platform.sfe.data_plane_cpu_allocation_max = cpu_max_allocation
+        elif self.shared_utils.is_wan_server:
+            # For AutoVPN Route Reflectors and Pathfinders, running on CloudEOS, setting
+            # this value is required for the solution to work.
+            msg = "For AutoVPN RRs and Pathfinders, 'data_plane_cpu_allocation_max' must be set"
+            raise AristaAvdInvalidInputsError(msg)
+
+        # populate interface profile for SFE platform (if supported)
+        if self.shared_utils.is_sfe_interface_profile_supported and (sfe_member_interfaces_for_profile := self._get_sfe_interface_profile_member_interfaces):
+            default_sfe_interface_profile_name = "Default_Interface_Profile"
+            # build single ProfilesItem and append to Profiles
+            self.structured_config.platform.sfe.interface.profiles.append_new(
+                name=default_sfe_interface_profile_name,
+                interfaces=sfe_member_interfaces_for_profile,
+            )
+            # indicate above profile as the one to apply for the device
+            self.structured_config.platform.sfe.interface.interface_profile = default_sfe_interface_profile_name
+
+    @cached_property
+    def _get_sfe_interface_profile_member_interfaces(
+        self: AvdStructuredConfigBaseProtocol,
+    ) -> EosCliConfigGen.Platform.Sfe.Interface.ProfilesItem.Interfaces | None:
+        """Returns list of eligible SFE interfaces with profile settings for structured config."""
+        # Iterate through all L3 interfaces checking for those with 'rx_queue' config
+        # Also iterate through member interfaces of all L3 Port-Channels with 'rx_queue' config
+        if not self.shared_utils.is_sfe_interface_profile_supported:
+            return None
```

Comment:
I dont think this check is needed as it should never be called if this is not true (the way it is used today, you can specify in docstring that the caller is in charge of checking this)